src/theme/interaction.rs

```diff
@@ -57,15 +62,55 @@ fn apply_interaction_palette(
     }
 }
 
+#[derive(Resource, Asset, Reflect, Clone)]
+pub struct ButtonAssets {
+    #[dependency]
+    hover: Handle<AudioSource>,
+    #[dependency]
+    press: Handle<AudioSource>,
+}
+
+impl ButtonAssets {
+    pub const PATH_BUTTON_HOVER: &'static str = "audio/sfx/button_hover.ogg";
+    pub const PATH_BUTTON_PRESS: &'static str = "audio/sfx/button_press.ogg";
+}
+
+impl FromWorld for ButtonAssets {
+    fn from_world(world: &mut World) -> Self {
+        let assets = world.resource::<AssetServer>();
+        Self {
+            hover: assets.load(Self::PATH_BUTTON_HOVER),
+            press: assets.load(Self::PATH_BUTTON_PRESS),
+        }
+    }
+}
+
 fn trigger_interaction_sfx(
     interaction_query: Query<&Interaction, Changed<Interaction>>,
+    button_assets: Res<ButtonAssets>,
     mut commands: Commands,
 ) {
     for interaction in &interaction_query {
         match interaction {
-            Interaction::Hovered => commands.play_sfx(SfxHandles::PATH_BUTTON_HOVER),
-            Interaction::Pressed => commands.play_sfx(SfxHandles::PATH_BUTTON_PRESS),
-            _ => (),
+            Interaction::Hovered => {
+                commands.spawn((
+                    AudioBundle {
+                        source: button_assets.hover.clone(),
+                        settings: PlaybackSettings::DESPAWN,
+                    },
+                    SoundEffect,
+                ));
+            }
+            Interaction::Pressed => {
+                commands.spawn((
+                    AudioBundle {
+                        source: button_assets.press.clone(),
+                        settings: PlaybackSettings::DESPAWN,
+                    },
+                    SoundEffect,
+                ));
+            }
+            _ => {}
```

Comment:
This could be simplified to only clone the handle within the `match` block, then have a single `commands.spawn` afterwards with the chosen handle.

Reply:
Good call!